### DISCUSSION TITLE
How to build on macOS?

### DISCUSSION THREAD

**Discussion by cgommel** [wmbusmeters/wmbusmeters#604] [General]
Hello,

after successfully building twmbusmeters i tried building it on macos (12.5.1 on apple silicon).

Unfortunately the fun came to an early end just during the ./configure command:

`configure: error: Could  not find rtlsdr library. Try: sudo apt install librtlsdr-dev`

I *did* install librtlsdr using homebrew but this did not improve the behavior:
`Warning: librtlsdr 0.6.0 is already installed and up-to-date.`

So what can I do to get it work?

**Comment by cgommel** [comment]
i figured out a workaround that seems to work for building.
```
export LDFLAGS="-L/opt/local/lib"   
export CXXFLAGS="-I/opt/local/include -std=c++14"
```

 `sudo make install` fails:
```
ld: library not found for -lrtlsdr
clang: error: linker command failed with exit code 1 (use -v to see invocation)

```

But the library exists:
```
chg@cgPro wmbusmeters % ls -lah /opt/local/lib/librtl*
-rwxr-xr-x  1 root  wheel   107K  4 Apr 21:28 /opt/local/lib/librtlsdr.0.6git.dylib
lrwxr-xr-x  1 root  wheel    22B  4 Apr 21:28 /opt/local/lib/librtlsdr.0.dylib -> librtlsdr.0.6git.dylib
-rw-r--r--  1 root  wheel    83K  4 Apr 21:28 /opt/local/lib/librtlsdr.a
lrwxr-xr-x  1 root  wheel    17B  4 Apr 21:28 /opt/local/lib/librtlsdr.dylib -> librtlsdr.0.dylib
```
So what can I do against?

**Reply by pbasista** [reply]
I was able to successfully proceed with `make` and `make install` after updating the `Makefile` in the following way:

```patch
diff --git a/Makefile b/Makefile
index 44e938d..6c9128a 100644
--- a/Makefile
+++ b/Makefile
@@ -140,6 +140,11 @@ ifeq ($(shell uname -s),FreeBSD)
     USBLIB    =  -lusb
 endif
 
+ifeq ($(shell uname -s),Darwin)
+    CXXFLAGS += -I/opt/homebrew/include
+    LDFLAGS  += -L/opt/homebrew/lib
+endif
+
 $(BUILD)/%.o: src/%.cc $(wildcard src/%.h)
        $(CXX) $(CXXFLAGS) $< -c -E > $@.src
        $(CXX) $(CXXFLAGS) $< -MMD -c -o $@
```
